src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -12688,5 +12687,327 @@ public void RefStructHandler_DynamicInHole(string expression)
+                // (4,12): error CS8944: 'lambda expression' is not an instance method, the receiver cannot be an interpolated string handler argument.
+                // var a =  ([InterpolatedStringHandlerArgument("")] CustomHandler c) => { };
+                Diagnostic(ErrorCode.ERR_NotInstanceInvalidInterpolatedStringHandlerArgumentName, @"InterpolatedStringHandlerArgument("""")").WithArguments("lambda expression").WithLocation(4, 12 + mod.Length)
+            );
+        }
+
+        [Theory]
+        [InlineData("static")]
+        [InlineData("")]
+        public void ArgumentsOnLambdas_02(string mod)
+        {
+            var code = @"
+using System;
+using System.Runtime.CompilerServices;
+
+var a = " + mod + @" (int i, [InterpolatedStringHandlerArgument(""i"")] CustomHandler c) => Console.WriteLine(c.ToString());
+
+a(1, $"""");
+
+partial struct CustomHandler
+{
+    public CustomHandler(int literalLength, int formattedCount, int i) => throw null;
+}
+";
+
+            var verifier = CompileAndVerify(new[] { code, InterpolatedStringHandlerArgumentAttribute, GetInterpolatedStringCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: false) }, expectedOutput: @"");
+            verifier.VerifyDiagnostics(
+                // (5,19): warning CS8971: InterpolatedStringHandlerArgument has no effect when applied to lambda parameters and will be ignored at the call site.
+                // var a =  (int i, [InterpolatedStringHandlerArgument("i")] CustomHandler c) => Console.WriteLine(c.ToString());
+                Diagnostic(ErrorCode.WRN_InterpolatedStringHandlerArgumentAttributeIgnoredOnLambdaParameters, @"InterpolatedStringHandlerArgument(""i"")").WithLocation(5, 19 + mod.Length)
+            );
+
+            verifier.VerifyIL("<top-level-statements-entry-point>", @"
+{
+  // Code size       45 (0x2d)
+  .maxstack  4
+  IL_0000:  ldsfld     ""System.Action<int, CustomHandler> Program.<>c.<>9__0_0""
+  IL_0005:  dup
+  IL_0006:  brtrue.s   IL_001f
+  IL_0008:  pop
+  IL_0009:  ldsfld     ""Program.<>c Program.<>c.<>9""
+  IL_000e:  ldftn      ""void Program.<>c.<<Main>$>b__0_0(int, CustomHandler)""
+  IL_0014:  newobj     ""System.Action<int, CustomHandler>..ctor(object, System.IntPtr)""
+  IL_0019:  dup
+  IL_001a:  stsfld     ""System.Action<int, CustomHandler> Program.<>c.<>9__0_0""
+  IL_001f:  ldc.i4.1
+  IL_0020:  ldc.i4.0
+  IL_0021:  ldc.i4.0
+  IL_0022:  newobj     ""CustomHandler..ctor(int, int)""
+  IL_0027:  callvirt   ""void System.Action<int, CustomHandler>.Invoke(int, CustomHandler)""
+  IL_002c:  ret
+}
+");
+        }
+
+        [Fact]
+        public void ArgumentsOnDelegateTypes_01()
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+
+M m = null;
+
+m($"""");
+
+delegate void M([InterpolatedStringHandlerArgument("""")] CustomHandler c);
```

Comment:
Does this fall out of the spec?

Reply:
There's nothing specific about lambdas in the spec. However, I don't think there needs to be in the interpolated strings proposal: there's nothing particularly different here than for any other attribute that affects the end result, such as `MaybeNull` or similar.